### ISSUE TITLE
No longer functions in Slack 3.0.0 beta

### ISSUE THREAD

**Issue by Bterrell721** [widget-/slack-black-theme#21]
Hello,

Thanks for this awesome package, been working wonderfully and I love it.  Unfortunately, Slack has recently updated with a new version that seems to invalidate it:

app-3.0.0-beta2638bae9b

I updated the new index.js file in the AppData path with the additional code and restarted Slack, but it seems to be entirely ignoring it and none of the styling or CSS is being applied.  Is there some additional tweak I need to make in order for this to continue to function?

Note that it still works perfectly in the most recent non-beta version of 2.8.2

Thanks again, and let me know if I can assist further.  Index file is attached.
[index.zip](https://github.com/widget-/slack-black-theme/files/1427874/index.zip)

**Comment by widget-** [comment]
Looks like they've re-organized how the the client gets loaded, which their changelog mentions as making everything faster and using less RAM, which is good.

In the current stable version, the "team" view (the part with the channels and chats) lived inside the main Electron panel, which was also where the sidebar is. It doesn't look like it's set up like that any more, so the `document.querySelectorAll(".TeamView webview");` selector doesn't find anything any more.

I looked for about 5 minutes and couldn't see how the views are loaded/organized now. You can try if you want, but I'll look later.

**Comment by brian000** [comment]
Same issue with `Beta 3.0.0-beta2 PR3895/638bae9 (Around The World)`

**Comment by SebScoFr** [comment]
Would love to see a patch for Slack3 as well :)

**Comment by Bterrell721** [comment]
Looks like they actually broke earlier version 2.8.2 now as well by changing the entire folder structure.  There is no longer a src folder, and the index file doesn't seem to exist anymore at all (see screen for reference).
![2017-11-06_11-05-35](https://user-images.githubusercontent.com/5151899/32453674-92718f60-c2e2-11e7-9d89-6b64837ce778.png)

**Comment by SamMarkGoldman** [comment]
I love this package and also can't wait to see support for Slack v3.

**Comment by skmexyz** [comment]
Adding this to `src/static/ssb-interop.js` appears to work for now.
```
document.addEventListener('DOMContentLoaded', function() {
  $.ajax({
   url: 'https://cdn.rawgit.com/laCour/slack-night-mode/master/css/raw/black.css',
   success: function(css) {
     $("<style></style>").appendTo('head').html(css);
   }
 });
});
```

**Comment by jrausch12** [comment]
@skmexyz where would you insert that code for a Debian system?

**Comment by Braunson** [comment]
@jrausch12 @skmexyz Same here, no path or file such as that on Windows either.

**Comment by avelis26** [comment]
Skmexyz's solution partially works for Slack3. The side bar is still purple but I can live with that.

**Comment by Oxxygen** [comment]
@avelis26 worth noting on OSX (and likely on Windows/Linux), the sidebar colors can be adjusted in preferences. For OSX its Slack -> Preferences ->Sidebar.

A sample sidebar dark theme could be the one found from [Dracula](https://draculatheme.com/slack/)

**Comment by Morydees** [comment]
This fix used to work on 3.00 but on 3.03 it seems to only partially work. One of my servers works perfectly, black background and white text but another has black on black and the text is not legible. Any idea why?

**Comment by skmexyz** [comment]
I'm on slack 3.0.2 and had to reapply the solution as above. Looks like autoupdates wipes your existing changes (as it should I guess).
Applying the above changes and restarting works just as before.

I combine the dark theme with Slack's Ochin theme for the sidebar to have a slightly more consistent Slack.

**Comment by Morydees** [comment]
@skmexyz yep, that matches my experience with 3.0.2 and I was very happy. You might skip 3.0.3 until we're sure on a solution there as I've been unable to get the old fix to work reliably. In fact, can you even disable automatic updating anymore? I sure can't seem to find it.

**Comment by tylercentral** [comment]
Looks like @skmexyz 's fix is working mostly ok again on version 3.0.4 on Windows 10: https://i.imgur.com/KaIH1tj.png

**Comment by Morydees** [comment]
I can't seem to find 3.0.4 on their site and in-program check for updates is showing I'm at latest version for 3.0.3. Perhaps staggered release?

**Comment by tylercentral** [comment]
Probably a staggered release. The slack for windows beta page shows `What’s new in 3.0.4` but there is no changelog for that release. I'm unsure, mine auto updated through the desktop app and decided to try this patch again to see how it functioned.

**Comment by jrausch12** [comment]
I recommend adding this to the readme. This is still a common issue for the latest slack as of 3.1.1 @widget-
